modules/processors/nih_dataset_processor.py

```diff
@@ -0,0 +1,59 @@
+import datasets
+import os
+from ..dataset_processor import *
+
+class NIHDataset(Processor):
+
+    def __init__(self, config_name='number',is_query=True, *args, **kwargs):
+        self.dataset_name='nih_v1_'+config_name
+        if is_query:
+            self.dataset_name+='_query'
+        else:
+            self.dataset_name+='_doc'
+        self.conf_name= config_name
+        self.is_query=is_query
+        
+        super().__init__(*args, **kwargs, dataset_name=self.dataset_name)
+    
+    def process(self):
+        print(self.conf_name)
```

Comment:
I checked the datasets, I'd personally be in favor of adding '(' around the needles, otherwise the documents don't really make a lot of sense. No ?

Reply:
Ok, I just took your functions  ( insert needle etc ...), maybe I did use it with the proper argument?

Reply:
It's just not present in my functions, but I've been wanting to add it for a while, it seems a bit treacherous to ask that to an LLM when the context does not really make sense. I'm ready to merge it as is anyway, your call.

Reply:
This is left for a future version of the datasets with more test ....